Mathlib/Algebra/GroupWithZero/WithZero.lean

```diff
@@ -35,6 +35,13 @@ instance one : One (WithZero α) where
 
 @[simp, norm_cast] lemma coe_one : ((1 : α) : WithZero α) = 1 := rfl
 
+-- We give this higher priority so that unzero_coe doesn't trigger.
+@[simp high] lemma unzero_one (hx : (1 : WithZero α) ≠ 0) : unzero hx = 1 := rfl
+
+lemma unzero_ofNat_one (hx : ((OfNat.ofNat 1) : WithZero α) ≠ 0) :
```

Comment:
```suggestion
lemma unzero_ofNat [n.AtLeastTwo] (hx : ofNat(n) : WithZero α) ≠ 0) :
```
was what I was thinking of. Maybe that's not valid though. 

Note also that you should never write `OfNat.ofNat 37`, as that means `OfNat.ofNat (OfNat.ofNat (nat_lit 37))`.

Reply:
I'm not sure if this is valid here. The only reason the `unzero_one` lemma above works is due to the instance [One.toOfNat1](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Notation/Defs.html#One.toOfNat1). In order for this lemma to work, we would need an `[OfNat α n]`, and `[OfNat (WithZero α) n]`, and presumably also something to say these are consistent, which I'm not sure exists.

Reply:
I'll delete the `ofNat` lemma